packages/api/models/executions.js

```diff
@@ -166,7 +167,20 @@ class Execution extends Manager {
       mutableFieldNames,
     });
 
-    await this.dynamodbDocClient.update(updateParams).promise();
+    // Do not allow out-of-order updates to set a completed or failed execution back to running
+    if (executionItem.status === 'running') {
+      updateParams.ConditionExpression = '(attribute_not_exists(status) or :status = #status)';
```

Comment:
`attribute_not_exists(status)` is illegal syntax for DynamoDB that results in the following error:
```
ValidationException (Error) {
    code: 'ValidationException',
    message: 'Invalid ConditionExpression: Attribute name is a reserved keyword; reserved keyword: status',
    requestId: '29a88069-42d9-40e3-872f-e63ef9cc6c3c',
    retryDelay: 31.21371676988891,
    retryable: false,
    statusCode: 400,
    time: Date 2021-03-31 17:50:11 590ms UTC {},
}
```

I'm not sure how to resolve this.